packages/remix/src/build.ts

```diff
@@ -70,6 +78,52 @@ export const build: BuildV2 = async ({
     }
   }
 
+  // Ensure `@remix-run/vercel` is in the project's `package.json`
+  const packageJsonPath = await walkParentDirs({
+    base: repoRootPath,
+    start: workPath,
+    filename: 'package.json',
+  });
+  console.log({ packageJsonPath });
+  if (packageJsonPath) {
+    const packageJson: PackageJson = JSON.parse(
+      await fs.readFile(packageJsonPath, 'utf8')
+    );
+    const { dependencies = {}, devDependencies = {} } = packageJson;
+
+    let modified = false;
+    if (REMIX_RUNTIME_ADAPTER_NAME in devDependencies) {
+      dependencies[REMIX_RUNTIME_ADAPTER_NAME] =
+        devDependencies[REMIX_RUNTIME_ADAPTER_NAME];
+      delete devDependencies[REMIX_RUNTIME_ADAPTER_NAME];
+      console.log(
+        `Warning: Moving "${REMIX_RUNTIME_ADAPTER_NAME}" from \`devDependencies\` to \`dependencies\`. You should commit this change.`
+      );
+      modified = true;
+    } else if (!('@remix-run/vercel' in dependencies)) {
+      dependencies[REMIX_RUNTIME_ADAPTER_NAME] = REMIX_RUNTIME_ADAPTER_VERSION;
+      console.log(
+        `Warning: Adding "${REMIX_RUNTIME_ADAPTER_NAME}" v${REMIX_RUNTIME_ADAPTER_VERSION} to \`dependencies\`. You should commit this change.`
+      );
+      modified = true;
+    }
+
+    if (modified) {
+      const packageJsonString = JSON.stringify(
+        {
+          ...packageJson,
+          dependencies,
+          devDependencies,
+        },
+        null,
+        2
+      );
+      await fs.writeFile(packageJsonPath, `${packageJsonString}\n`);
```

Comment:
Will this break lock files?

Reply:
Shouldn't. `npm` updates the `package-lock.json` file after this update, at least.